client/blocks/image-selector/README.md

```diff
@@ -0,0 +1,61 @@
+Image Selector
+===
+
+Image Selector is a component that uses the Media Modal to select images and preview previously selected images.  It allows multiple images to be passed or it can be limited to a single item. 
+
+## Usage
+
+Pass an array of image IDs that will be shown in the preview and preselected in the Media Modal.
+
+Changes to the selection of images will be passed back in various props for setting images, dropping images, and removing images.  See the example.jsx file for a more complete example.
+
+```jsx
+import ImageSelector from 'blocks/image-selector';
+
+export default function MyFeaturedImages() {
+	return (
+		<ImageSelector imageIds={ [ 34, 105 ] } multiple />
+	);
+}
+```
+
+This component needs both its own styles imported as well as the Media Modal styles on the page in which it's displayed.
+
+```scss
+@import 'assets/stylesheets/sections/media';
+@import 'blocks/image-selector/style';
+```
+
+### Props
+
+Props are displayed as a table with Name, Type, Default, and Description as headings.
+
+Name | Type | Default | Description
+--- | --- | --- | ---
+`className` | `string` | `` | String used for the class of outer ImageSelector wrapper
+`compact` | `bool` | `0` | Determines whether or not the upload picker is compact or full size
+`hasDropZone` | `bool` | `0` | Enables dropzone for drag-and-drop images
+`imageIds` | `array` | `[]` | An array of image IDs to be shown in the preview and selected in the Media Modal
+`maxWidth` | `number` | `450` | Maximum width to be used for a preview image
+`onAddImage` | `func` | `noop` | Passes a single transient image object after an image is dropped on the DropZone
+`onImageChange` | `func` | `noop` | Passes the entire array of currently selected image objects when transitioning from a transient to permanent image
+`onImageSelected`* | `func` | `noop` | Passes an array of selected image objects when set from the Media Modal
```

Comment:
Does the `*` mean that the prop is required? If yes, shouldn't we also mark it as required in the `propTypes`, too?

Reply:
Absolutely, good call.  My intention originally was to not require any of these props, but I can't come up with any use cases currently where we wouldn't want to handle images when passed back.